src/compiler/checker.ts

```diff
@@ -3993,6 +4027,37 @@ namespace ts {
             return links.resolvedType;
         }
 
+        // We do not perform supertype reduction on intersection types. Intersection types are created only by the &
+        // type operator and we can't reduce those because we want to support recursive intersection types. For example,
+        // a type alias of the form "type List<T> = T & { next: List<T> }" cannot be reduced during its declaration.

```

Comment:
Yes, getApparentType should (and does) trigger subtype reduction. Even though subtype reduction should just be an optimization, it's effects are actually observable in a few cases. For example, expression operators don't apply to `number | number`, but do apply to the reduced type `number`. And call signatures are less restricted when a union type reduces to a single type. So, technically we should spec exactly when we perform subtype reduction.